### PR TITLE
Initial draft of application context

### PR DESCRIPTION
**Pull request by duncan**
As discussed in #136, an initial draft of an application context with `name`, `version`, `build`, `androidFlavor`, and `buildIsRelease` properties.

### CODE REVIEW

File: schemas/com.snowplowanalytics.snowplow/application_context/jsonschema/1-0-0 [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "name": {
+      "type": "string"
+    },
+    "version": {
+      "type": "string"
+    },
+    "build": {
+      "type": "string"
+    },
+    "androidFlavor": {
```
Comment (alexanderdean) [inline schemas/com.snowplowanalytics.snowplow/application_context/jsonschema/1-0-0]: This was going to be buildFlavor no?

Comment (duncan) [inline schemas/com.snowplowanalytics.snowplow/application_context/jsonschema/1-0-0]: 😳 yah. it was. Fixing.

**Comment by duncan** [comment]
I guess the other thing one could call it that might be slightly more generic would be `buildVariant`... ?

**Comment by alexanderdean** [comment]
Oh yah, I like `buildVariant`.

**Comment by alexanderdean** [comment]
Coming back to this cold 6 months later, "build" and "buildVariant" feel a bit unintuitive.

**Comment by alexanderdean** [comment]
Hey @jbeemster - the original genesis for this context was Android apps. Would it be possible to automatically generate this context for an Android app, or would it need to be manually set by the app author? What about for Objective-C?

**Comment by jbeemster** [comment]
It would need to be setup as a Tracker option but all that information should be available for both Android and Objective-C.  I am thinking it might be worthwhile exposing a Context object of sorts which handles all of the various contexts that we have popping up..

**Comment by alexanderdean** [comment]
Sounds good - can you create tickets for both mobile trackers then @jbeemster ?

**Comment by alexanderdean** [comment]
From a Java/Scala/Clojure perspective, the obvious field missing is organization aka vendor.

**Comment by alexanderdean** [comment]
Consider grouping the build fields into a nested sub-object

**Comment by TiernanKennedy** [comment]
Spec looks good to me @alexanderdean.

With iOS we always have a build number, so requiring that would be fine. 

I'm not sure why buildVariant is needed, given that we have the mobile_context?

**Comment by alexanderdean** [comment]
Thanks @TiernanKennedy. I think `buildVariant` is an Android thing?

**Comment by TiernanKennedy** [comment]
Hmm, should there be OS specific stuff in here? Part of me thinks there should just be the addition of `build` and `version` in the mobile context. Is there a reason to keep them separate? @alexanderdean

**Comment by alexanderdean** [comment]
The mobile context is intended to describe the mobile device which is hosting the app; the application context is intended to describe the app which is running the tracker - so application build and version details belong in the application context. We could potentially rename `buildVariant` to something less Android-flavoured...

**Comment by alexanderdean** [comment]
Related: http://stackoverflow.com/questions/30223676/adding-a-build-variant-and-running-it-on-ios-8

**Comment by TiernanKennedy** [comment]
Ah I see what you mean now. Yeah then `buildVariant` does make sense.

:+1:

**Comment by alexanderdean** [comment]
We're missing the application license as well...

**Comment by alexanderdean** [comment]
Should we add sourceRepository, sourceTag and sourceCommit too?

**Comment by bogaert** [comment]
> We're missing the application license as well...
> Should we add sourceRepository, sourceTag and sourceCommit too?

I don't think we need either for a first release of the schema.

**Comment by snowplowcla** [comment]
@duncan has signed the Software Grant and Corporate Contributor License Agreement

**Comment by alexanderdean** [comment]
I'm not sure what I meant by application license.

> >  Should we add sourceRepository, sourceTag and sourceCommit too?
>
> I don't think we need either for a first release of the schema.

Because of *Redshift blocked schemas*, the first release will be the only release for the foreseeable future.

**Comment by chuwy** [comment]
Pushing back to R97. @keanerobinson please let us know what do you think of it when you're back.

**Comment by chuwy** [comment]
Closing in favor of https://github.com/snowplow/iglu-central/pull/847

**Pull request closed without merging**